WooCommerce/Classes/POS/ViewModels/PointOfSaleDashboardViewModel.swift

```diff
@@ -106,12 +110,40 @@ final class PointOfSaleDashboardViewModel: ObservableObject {
     func cardPaymentTapped() {
         Task { @MainActor in
             showsCreatingOrderSheet = true
-            let order = try await createTestOrder()
+            let order = try await createTestOrder(amount: formattedOrderTotalPrice ?? "15.00")
             showsCreatingOrderSheet = false
-            let _ = try await cardPresentPaymentService.collectPayment(for: order, using: .bluetooth)
+            try await collectPayment(for: order)
+        }
+    }
+
+    @MainActor
+    private func collectPayment(for order: Order) async throws {
+        let _ = try await cardPresentPaymentService.collectPayment(for: order, using: .bluetooth)
+
+        // TODO: Here we should present something to show the payment was successful or not,
+        // and then clear the screen ready for the next transaction.
+    }
+
+    @MainActor
+    func totalsAppeared() async {
+        // Digging in to the connection viewmodel here is a bit of a shortcut, we should improve this.
+        // TODO: Have our own subscription to the connected readers
+        if cardReaderConnectionViewModel.connectionStatus == .connected {
+            // Once we have finalised order creation/update, this check shouldn't be required.
+            // Instead, we should have whatever code does the order creation/update kick off this connection process.
+            guard let order = try? await createOrUpdateTestOrder() else {
```

Comment:
❓ How about calling `createOrUpdateTestOrder` at the beginning of this function (when the totals view appears) or when the merchant taps to check out regardless of the reader connection state? I think we'd like to create/update the order remotely as soon as the merchant checks out so that the order totals can be shown ASAP. In the current implementation, related to the ❓ comment above, the order isn't created right away when the reader is disconnected.

Reply:
I had it there initially... but because this PR is focused on the connected-reader flow, I didn't want to interfere with the disconnected reader flow here. 

If we do it right at the start of this function, we could use the presence of an order to enable/disable the Collect Payment button, but I'd rather leave that until Bozidar's real order creation is done.

By keeping it here, it only happens when we are going to prepare a reader, and can be separately triggered for button taps (as in 034cbcb)